### PR TITLE
python311Packages.protoc-gen-swagger: init at b572618d0aadcef63224bf85ebba05270b573a53

### PR DESCRIPTION
**Pull request by akshatagarwl**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/protoc-gen-swagger/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+, fetchFromGitHub
+, pytestCheckHook
+, protobuf
+, pythonOlder
+, setuptools
+}:
+
+buildPythonPackage rec {
+  pname = "protoc-gen-swagger";
+  version = "b572618d0aadcef63224bf85ebba05270b573a53";
```
Comment (natsukium) [inline pkgs/development/python-modules/protoc-gen-swagger/default.nix:12]: This commit has been tagged as `0.1.0`.

File: pkgs/development/python-modules/protoc-gen-swagger/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "universe-proton";
+    repo = pname;
```
Comment (natsukium) [inline pkgs/development/python-modules/protoc-gen-swagger/default.nix:17]:
```suggestion
    repo = "proto-gen-swagger";
```

File: pkgs/development/python-modules/protoc-gen-swagger/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    rev = version;
+    hash = "sha256-ooY0W+bgkmrI4lvqile4/4016LJsx5c6gc4rnVmNtVU=";
+  };
+
+  propagatedBuildInputs = [
+    setuptools
```
Comment (natsukium) [inline pkgs/development/python-modules/protoc-gen-swagger/default.nix:23]: `setuptools` should be go to `nativeBuildInputs`.

**Review by natsukium — COMMENTED** [review]
The target version of `grpc-gateway` looks quite different. Is it still a valid package?

**Comment by humancalico** [comment]
> The target version of grpc-gateway looks quite different. Is it still a valid package?

I needed this to package https://github.com/flyteorg/flytekit/. I now see that [protoc-gen-swagger has been renamed protoc-gen-openapiv2](https://github.com/grpc-ecosystem/grpc-gateway/blob/7da715e510ff64375596028d4e5108d6255ceb01/docs/docs/development/grpc-gateway_v2_migration_guide.md#protoc-gen-swagger-has-been-renamed-protoc-gen-openapiv2). Let me check if this is actually being used in the upstream repository.

**Pull request closed without merging**